java/vector/src/main/java/org/apache/arrow/vector/util/ByteFunctionHelpers.java

```diff
@@ -49,6 +49,45 @@ public static final int equal(final ArrowBuf left, int lStart, int lEnd, final A
     return memEqual(left.memoryAddress(), lStart, lEnd, right.memoryAddress(), rStart, rEnd);
   }
 
+  /**
+   * Compute hashCode with the given {@link ArrowBuf} and start/end index.
+   */
+  public static final int hash(final ArrowBuf buf, int start, int end) {
+    long addr = buf.memoryAddress();
+    int len = end - start;
+    long pos = addr + start;
+
+    int hash = 0;
+
+    while (len > 7) {
+      long value = PlatformDependent.getLong(pos);
+      hash = comebineHash(hash, Long.hashCode(value));
+
+      pos += 8;
+      len -= 8;
+    }
+
+    while (len > 3) {
+      int value = PlatformDependent.getInt(pos);
+      hash = comebineHash(hash, Integer.hashCode(value));
```

Comment:
this is a no-op I believe.

Reply:
Thanks for comments, IMO, what dose this mean? :)

Reply:
@emkornfield Could you please explain this a little more? thanks

Reply:
I think Integer.hashCode(value) == value?

Reply:
I see, you are right, thanks, fixed.